app/controllers/api/base_controller/action.rb

```diff
@@ -16,13 +16,13 @@ def api_action(type, id)
         result
       end
 
-      # wrapper around api_action for creating a resource
-      # alternative to api_resource when creating a resource
+      # wrapper around api_action for acting on an ems resource
+      # alternative to api_resource when working on an ems.
       #
       # NOTE: would have been nice to only pass ems_id, and name
       #       That way the external interface would be not this method's concern
       #       but passing data also allows asserting that an id is not passed
-      def create_ems_resource(type, data, supports: false)
+      def api_ems_resource(type, data, action_phrase, supports: false)
```

Comment:
(If you do not mind, I'll add some comments since I've checked it on my side)

As I see, you don't use the parameter action_phrase inside this function.. so when you're calling this function in 'create_ems_resource' it sends nil to argument 'data', and there is an extra argument.

So just need to change the #{action} to #{action_phrase} in api_ems_resource and remove the 'nil' parameter when we call this function in 'create_ems_resource'.

Reply:
thank you.
I had seen the `id` there before, so I added it into the api but then realized that you did not need the id (it was `nil`) so I removed it from one spot but forgot to remove from another.

Let me know if this works closer to your goals